docs/examples/rx/03_listen_to_balance_change/index.js

```diff
@@ -11,14 +13,20 @@ async function main () {
   let previous = null;
 
   // Here we pass the (optional) provider
-  ApiRx.create(wsProvider).subscribe((api) => {
-    console.log(`${Alice} has ${previous || '???'} previous balance`);
-    console.log(`You may leave this example running and start example 06 ` +
-                `or transfer any value to Alice at ${Alice}`);
-
-    // Here we subscribe to any balance changes and update the on-screen value.
-    // Use the Storage chain state (runtime) Node Interface.
-    api.query.balances.freeBalance(Alice).subscribe((current) => {
+  // We avoid having nested `.subscribe` blocks. Use `.subscribe` or
+  // `.tap` blocks for debugging with console.log
+  ApiRx
+    .create(wsProvider)
+    .pipe(
+      // Here we subscribe to any balance changes and update the on-screen value.
+      // Use the Storage chain state (runtime) Node Interface.
+      switchMap((api) => api.query.balances.freeBalance(Alice))
+    )
+    .subscribe((current) => {
+      console.log(`${Alice} has ${previous || '???'} previous balance`);
+      console.log(`You may leave this example running and start example 06 ` +
+                  `or transfer any value to Alice at ${Alice}`);
+
       // Calculate the delta
       const change = current.sub(previous);
```

Comment:
There's a way to not have the external `previous` variable, using the rxjs `scan` operator. But that might create confusion for beginners, so let's leave it like this.

Reply:
I'll add that as a "challenge"